src/log.c

```diff
@@ -101,15 +100,24 @@ void fi_log_init(void)
 {
 	struct fi_filter subsys_filter;
 	int level, i;
+	char *levelstr = NULL, *provstr = NULL, *subsysstr = NULL;
 
-	level = fi_read_value("FI_LOG_LEVEL", log_levels);
+	fi_param_define(NULL, "log_level", FI_PARAM_STRING,
+			"Specify logging level: warn, trace, info, debug (default: warn)");
+	fi_param_get_str(NULL, "log_level", &levelstr);
```

Comment:
> > +	fi_param_define(NULL, "log_level", FI_PARAM_STRING,
> > +			"Specify logging level: warn, trace, info, debug
> (default: warn)");
> > +	fi_param_get_str(NULL, "log_level", &levelstr);
> 
> In my original patch there was error checking + warning here. I think it
> is OK to not do it, but if something goes wrong there is no indication
> (and won't this trigger gcc warnings?). This is where that one-shot API
> that I talked about in #1112

The chances of a failure here are very low.  But this will produce a warning when fi_param_get_str is called for an undefined parameter.